mnemosyne/core/config.py

```
@@ -0,0 +1,487 @@
+        for key in changed:
+            config_key = self._yaml_to_config_key(key)
+            if config_key and config_key in REQUIRES_RESTART:
+                logger.warning(
+                    "Config key '%s' requires restart to take effect. "
+                    "The new value will apply on next process start.",
+                    config_key,
+                )
+
+        return changed
+
+    def get(self, key: str, default: Any = None) -> Any:
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `default`

(ANN401)